src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -191,35 +191,43 @@ public DateTime(int year, int month, int day, Calendar calendar)
         //
         public DateTime(int year, int month, int day, int hour, int minute, int second)
         {
-            if (second == 60 && s_systemSupportsLeapSeconds && IsValidTimeWithLeapSeconds(year, month, day, hour, minute, second, DateTimeKind.Unspecified))
+            if (second != 60 || !s_systemSupportsLeapSeconds)
             {
-                // if we have leap second (second = 60) then we'll need to check if it is valid time.
-                // if it is valid, then we adjust the second to 59 so DateTime will consider this second is last second
-                // in the specified minute.
-                // if it is not valid time, we'll eventually throw.
-                second = 59;
+                _dateData = DateToTicks(year, month, day) + TimeToTicks(hour, minute, second);
+            }
+            else
+            {
+                // if we have a leap second, then we adjust it to 59 so that DateTime will consider it the last in the specified minute.
+                WithLeapSecond(out this, year, month, day, hour, minute);
```

Comment:
>WithLeapSecond [](start = 16, length = 14)

why can't just do 
```C#
_dateData = DateToTicks(year, month, day) + TimeToTicks(hour, minute, second);
```

and then call VlidateLeapSeconds method on `this` after that? the current pattern looks a little bit strange to me.

Reply:
This was to save some prolog overhead for the common case, but it turns out to be a quite small difference, so removed these wrapper methods.